packages/ddp-server/livedata_server.js

```diff
@@ -1718,8 +1718,19 @@ DDPServer._calculateVersion = calculateVersion;
 // "blind" exceptions other than those that were deliberately thrown to signal
 // errors to the client
 var wrapInternalException = function (exception, context) {
-  if (!exception || exception instanceof Meteor.Error)
+  if (!exception) return exception;
+
+  // To allow packages to throw errors intended for the client but not have to
+  // depend on the Meteor.Error class, `isClientSafe` can be set to true on any
+  // error before it is thrown.
+  if (exception.isClientSafe) {
+    if (!(exception instanceof Meteor.Error)) {
+      const originalMessage = exception.message;
+      exception = new Meteor.Error(exception.error, exception.reason, exception.details);
```

Comment:
So in `Meteor.Error` constructor, it would `if (error instanceof Error)` then copy all "own" props plus message and stack to self?

I like that idea in general, but as I look at this a bit more, I'm not sure that casting to `Meteor.Error` here is actually necessary. Whatever this function returns is set to the `error` prop on the DDP payload, and `stringifyDDP` seems to just clone and send it. `addType` is not even used for `Meteor.Error` for EJSON, so it might just work with any error already?

However, I think the client does assume there will be error, reason, and details props in reconstructing the error on the other end.